### PR TITLE
Make sure the PM CC Triage Team has Triage Access to Public Repos

### PR DESCRIPTION
**Pull request by feanil**
- feat: Add a new generic check for a team.
- feat: Add an `is_relevant` interface.
- feat: Use the RequireTeamPermission check in the CLA Check.
- feat: Add a CC Triage Team Check.

Note for reviewers: This one will probably be easier to review commit by commit.

### CODE REVIEW

File: migrate/repo_checks.py
Comment (kdmccormick) [inline migrate/repo_checks.py:85]: I think extracting this class was a good refactoring, but the docstring confuses me. It seems like it'd be fine to initialize and use this class directly; in fact you do that below in `RequireCLACheck`.

File: migrate/repo_checks.py
Comment (kdmccormick) [inline migrate/repo_checks.py:87]: Consider documenting the types / possible values of these arguments, either via type annotations or just a docstring. I figured out that `permission` was a string, but it took me a bit to get there, and I still don't know what values it could accept.

**Review by kdmccormick — COMMENTED** [review]
Looking good, just two documentation suggestions.

I haven't tested it out yet but I'll do that first thing Monday morning.

**Review by kdmccormick — COMMENTED** [review]
I ran this today and it seemed to work great. A few general comments on this new check-and-fix framework since I didn't review the initial PR.. none of these are blockers for this PR:

* I think this is a good solution for repository management. As much as I would have loved if Terraform just worked, I think there's too much complexity in GitHub for us to successfully use Terraform alongside people manually editing settings. This seems like a great compromise.
* The output should tell me whether I'm doing a dry run or not. I blindly ran `python migrate/repo_checks.py` and freaked out a bit when I starting seeing the `Added triage access for...` log message, and finally calmed down when I read the code and saw that `--dry-run` is True by default.
* Before you/we start regularly using this to make real changes against the org, could you move it out of the `migrate` folder and document it in the README? We should probably rename this repo too. (edit: better yet, IMO this script would make sense in [repo-tools](https://github.com/openedx/repo-tools/), which'd let us archive this repository)

File: migrate/repo_checks.py
Comment (nedbat) [inline migrate/repo_checks.py:469]: You might like [`textwrap.indent`](https://docs.python.org/3/library/textwrap.html#textwrap.indent).

File: migrate/repo_checks.py
Comment (kdmccormick) [inline migrate/repo_checks.py:82]:
```suggestion
    """
    Require that a team has a certain level of access to a repository.

    To use this class as a check, create a subclass that specifies a particular
    team and permission level, such as RequireTriageTeamAccess below.
    """
```
The current docstring states your opinion on what this class is for, but it doesn't tell the reader what the class actually does or how they should use it.

File: migrate/repo_checks.py
Comment (kdmccormick) [inline migrate/repo_checks.py:153]: expand acronyms & avoid Jira terminology
```suggestion
    The Core Contributor Triage Team needs to be able to triage
    issues in all repos in the Open edX Platform.
```

File: migrate/repo_checks.py
Comment (kdmccormick) [inline migrate/repo_checks.py:53]: Nice.

**Review by kdmccormick — APPROVED** [review]
LGTM pending these two suggestions. Thanks for breaking it up by commit.

Also, in your commit message body please expand CCs to Core Contributors.

File: migrate/repo_checks.py
Comment (nedbat) [reply migrate/repo_checks.py:153]: FWIW, Jira's term is also "issue".  :)

**Pull request merged**
